pygmt/src/plot.py

```diff
@@ -209,6 +209,14 @@ def plot(self, x=None, y=None, data=None, size=None, direction=None, **kwargs):
     extra_arrays = []
     if "S" in kwargs and kwargs["S"][0] in "vV" and direction is not None:
         extra_arrays.extend(direction)
+    elif (
+        "S" not in kwargs
+        and data is not None
+        and data is object
+        and "_geometry_column_name" in data.__dict__
+        and (data.geom_type == "Point").all()
```

Comment:
While this works, I do have two questions:

1. Will MultiPoint geometry types work?
2. Do you think it's ok to set the default Point style if **all** rows are points? I have concerns about how fast the check is say on 1000 rows (it may well be fast but I'm not sure).

Reply:
1. the current implementation works only with Point, I don't think we should support mixed geometrys in geopandas DataFrame it's a bad practice. we can add support for MultiPoint alon with `(data.geom_type == "Point").all() or (data.geom_type == "MultiPoint").all()`
2.  We don't need to worry about the execution time of `(data.geom_type == "Point").all()` because the execution time of writing the temporary file is larger (probably also plotting the figure but I didn't check)
from:
`import numpy as np
import pandas as pd
import geopandas as gpd
import timeit

x = np.random.uniform(0, 10, 1000)
y = np.random.uniform(0, 10, 1000)
df = pd.DataFrame(dict(x=x, y=y))
gdf = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.x, df.y))

num = 1000
print(timeit.timeit(lambda: (gdf.geom_type == 'Point').all(), number=num) / num)
print(timeit.timeit(lambda: gdf.to_file(filename='ttt'), number=num) / num)`

we get:
0.0010915064080000007
0.055181070641